eclass/cmake.eclass

```diff
@@ -343,25 +370,36 @@ cmake_src_prepare() {
 	fi
 
 	local modules_list
-	if [[ $(declare -p CMAKE_REMOVE_MODULES_LIST) == "declare -a"* ]]; then
-		modules_list=( "${CMAKE_REMOVE_MODULES_LIST[@]}" )
-	else
+	if [[ ${EAPI} == 7 && $(declare -p CMAKE_REMOVE_MODULES_LIST) != "declare -a"* ]]; then
 		modules_list=( ${CMAKE_REMOVE_MODULES_LIST} )
+	else
+		modules_list=( "${CMAKE_REMOVE_MODULES_LIST[@]}" )
 	fi
 
 	local name
 	for name in "${modules_list[@]}" ; do
-		find "${S}" -name ${name}.cmake -exec rm -v {} + || die
+		if [[ ${EAPI} == 7 ]]; then
+			find "${S}" -name ${name}.cmake -exec rm -v {} + || die
+		else
+			find -name "${name}.cmake" -exec rm -v {} + || die
+		fi
 	done
 
 	# Remove dangerous things.
 	_cmake_modify-cmakelists
 
-	popd > /dev/null || die
+	if [[ ${EAPI} == 7 ]]; then
+		popd > /dev/null || die
+	fi
 
-	# make ${S} read-only in order to detect broken build-systems
+	# Make ${CMAKE_USE_DIR} (in EAPI-7: ${S}) read-only in order to detect
+	# broken build systems.
 	if [[ ${CMAKE_QA_SRC_DIR_READONLY} && ! ${CMAKE_IN_SOURCE_BUILD} ]]; then
-		chmod -R a-w "${S}"
+		if [[ ${EAPI} == 7 ]]; then
```

Comment:
I wonder if it wouldn't make sense to do this ```if``` once at the beginning of the function and assign it to a local variable and just use that everywhere throughout the function without any further EAPI switching.

Reply:
I don't see much benefit there.